nifi-commons/nifi-security-utils/src/test/groovy/org/apache/nifi/security/util/SslContextFactoryTest.groovy

```diff
@@ -0,0 +1,272 @@
+import java.security.Security
+
+@RunWith(JUnit4.class)
+class SslContextFactoryTest extends GroovyTestCase {
+    private static final Logger logger = LoggerFactory.getLogger(SslContextFactoryTest.class)
+
+    private static final String KEYSTORE_PATH = "src/test/resources/TlsConfigurationKeystore.jks"
+    private static final String KEYSTORE_PASSWORD = "keystorepassword"
+    private static final String KEY_PASSWORD = "keypassword"
+    private static final KeystoreType KEYSTORE_TYPE = KeystoreType.JKS
+
+    private static final String TRUSTSTORE_PATH = "src/test/resources/TlsConfigurationTruststore.jks"
+    private static final String TRUSTSTORE_PASSWORD = "truststorepassword"
+    private static final KeystoreType TRUSTSTORE_TYPE = KeystoreType.JKS
+
+    private static final String PROTOCOL = CertificateUtils.CURRENT_TLS_PROTOCOL_VERSION
+
+    private static final Map<String, String> DEFAULT_PROPS = [
+            (NiFiProperties.SECURITY_KEYSTORE)         : KEYSTORE_PATH,
+            (NiFiProperties.SECURITY_KEYSTORE_PASSWD)  : KEYSTORE_PASSWORD,
+            (NiFiProperties.SECURITY_KEY_PASSWD)       : KEY_PASSWORD,
+            (NiFiProperties.SECURITY_KEYSTORE_TYPE)    : KEYSTORE_TYPE.getType(),
+            (NiFiProperties.SECURITY_TRUSTSTORE)       : TRUSTSTORE_PATH,
+            (NiFiProperties.SECURITY_TRUSTSTORE_PASSWD): TRUSTSTORE_PASSWORD,
+            (NiFiProperties.SECURITY_TRUSTSTORE_TYPE)  : TRUSTSTORE_TYPE.getType(),
+    ]
+
+    private NiFiProperties mockNiFiProperties = NiFiProperties.createBasicNiFiProperties("", DEFAULT_PROPS)
+
+    private TlsConfiguration tlsConfiguration
+
+    @BeforeClass
+    static void setUpOnce() throws Exception {
+        Security.addProvider(new BouncyCastleProvider())
+
+        logger.metaClass.methodMissing = { String name, args ->
+            logger.info("[${name?.toUpperCase()}] ${(args as List).join(" ")}")
+        }
+    }
+
+    @Before
+    void setUp() {
+        tlsConfiguration = new TlsConfiguration(KEYSTORE_PATH, KEYSTORE_PASSWORD, KEY_PASSWORD, KEYSTORE_TYPE, TRUSTSTORE_PATH, TRUSTSTORE_PASSWORD, TRUSTSTORE_TYPE)
+    }
+
+    @After
+    void tearDown() {
+
+    }
+
+    @Test
+    void testShouldCreateSslContextFromTlsConfiguration() {
+        // Arrange
+        logger.info("Creating SSL Context from TLS Configuration: ${tlsConfiguration}")
+
+        // Act
+        SSLContext sslContext = SslContextFactory.createSslContext(tlsConfiguration, SslContextFactory.ClientAuth.NONE)
+        logger.info("Created SSL Context: ${KeyStoreUtils.sslContextToString(sslContext)}")
+
+        // Assert
+        assert sslContext.protocol == tlsConfiguration.protocol
+
+        def defaultSSLParameters = sslContext.defaultSSLParameters
+        logger.info("Default SSL Parameters: ${KeyStoreUtils.sslParametersToString(defaultSSLParameters)}" as String)
+        assert defaultSSLParameters.getProtocols() == ["TLSv1.2", "TLSv1.1", "TLSv1"] as String[]
```

Comment:
```suggestion
        assert defaultSSLParameters.getProtocols() as Set == ["TLSv1.2", "TLSv1.1", "TLSv1"] as Set
```

Reply:
It's interesting that these are failing on Java 8 due to order. I didn't use `Set`s in this case because I _did_ want to enforce the order of these protocol versions. In Java 11, this is the consistent order that's returned. I'll update these tests to enforce the assertions differently based on the available JRE version. Thanks.